### ISSUE TITLE
[Bug] Unweighted Cross-Entropy on 95:5 Class Imbalance Causes Complete Brake Neglect

### ISSUE THREAD

**Issue by RustamSheoran** [Mohammedmt10/self-driving-model-in-simulation#5]
## Labels
`bug`, `critical`, `safety`, `loss-function`

## Affected Files
- `model.py` (lines 311, 349)

## Problem Summary
The brake head is framed as a 2-class classifier and trained using standard unweighted `nn.CrossEntropyLoss()`.

## Root Cause Code
```python
# model.py line 311 & 349
ce_criterion = nn.CrossEntropyLoss()
...
l_brake = ce_criterion(brake_logits, targets[:, 2].long())
```

## Why It Is Completely Ruining Everything
In CARLA autonomous driving datasets, braking occurs in only ~3%–5% of recorded frames (`control_brake = 1`). Under unweighted cross-entropy, a degenerate network that predicts `brake = 0` (no brake) 100% of the time achieves **95%–97% accuracy** with negligible loss. 

The author misdiagnosed this failure as "overfitting" and attempted to fix it with heavy weight decay (`1e-3`) and 50% dropout, which only made it harder for the network to activate the brake. As a result, the model refuses to brake for stopped vehicles, red lights, or pedestrians.

## Proposed Fix
Use class-weighted CrossEntropyLoss or Focal Loss to heavily penalize false negatives:
```python
# With 95% negative, 5% positive:
class_weights = torch.tensor([1.0, 19.0], device=device)
ce_criterion = nn.CrossEntropyLoss(weight=class_weights)
```
